activerecord/test/cases/migration_test.rb

```diff
@@ -309,6 +309,21 @@ def migrate(x)
         "On error, the Migrator should revert schema changes but it did not."
     end
 
+    def test_allows_sqlite3_rollback_on_invalid_column_type
+      if current_adapter?(:SQLite3Adapter)
+        Person.connection.create_table :something, force: true do |t|
```

Comment:
@rafaelfranca Have added your changes to the code below.  Should I nest this in the if block where I have it currently as @maclover7 suggested:
```Ruby
#/activerecord/test/cases/migration_test.rb:269
if ActiveRecord::Base.connection.supports_ddl_transactions?
```
or make it a new if block at the same level as the other similar if blocks on lines 554, 591, & 605. 

```Ruby
   #/activerecord/test/cases/migration_test.rb:312
if current_adapter?(:SQLite3Adapter)
  def test_allows_sqlite3_rollback_on_invalid_column_type
    Person.connection.create_table :something, force: true do |t|
      t.column :number, :integer
      t.column :name, :string
      t.column :foo, :bar
    end
    assert Person.connection.column_exists?(:something, :foo)
    assert_nothing_raised { Person.connection.remove_column :something, :foo, :bar }
    assert !Person.connection.column_exists?(:something, :foo)
    assert Person.connection.column_exists?(:something, :name)
    assert Person.connection.column_exists?(:something, :number)
  ensure
    Person.connection.drop_table :something, if_exists: true
  end
end 
```